homeassistant/components/paperless_ngx/binary_sensor.py

```diff
@@ -0,0 +1,122 @@
+"""Binary sensor support for Paperless-ngx."""
+
+from __future__ import annotations
+
+from _collections_abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import (
+    DOMAIN,
+    ENTITY_ATTRIBUTE_LAST_CHECKED,
+    ENTITY_ATTRIBUTE_LATEST_VERSION,
+    ENTITY_BINARYSENSOR_UPDATE_AVAILABLE,
+)
+from .coordinator import (
+    PaperlessConfigEntry,
+    PaperlessCoordinator,
+    PaperlessData,
+    PaperlessRuntimeData,
+)
+from .entity import PaperlessCoordinatorEntity
+from .helpers import build_state_fn
+
+
+@dataclass(frozen=True, kw_only=True)
+class PaperlessBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Paperless-ngx binary sensor entity."""
+
+    value_fn: Callable[[PaperlessData], bool | None]
+    attributes_fn: Callable[[PaperlessData], dict[str, str | None]] | None = None
+
+
+BINARY_SENSOR_DESCRIPTIONS: tuple[PaperlessBinarySensorEntityDescription, ...] = (
+    PaperlessBinarySensorEntityDescription(
+        key=ENTITY_BINARYSENSOR_UPDATE_AVAILABLE,
+        translation_key=ENTITY_BINARYSENSOR_UPDATE_AVAILABLE,
+        device_class=BinarySensorDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=build_state_fn(
+            lambda data: data.remote_version.update_available
+            if data.remote_version is not None
+            else None,
+        ),
+        attributes_fn=lambda data: {
+            ENTITY_ATTRIBUTE_LATEST_VERSION: str(data.remote_version.version)
+            if data.remote_version is not None
+            else None,
+            ENTITY_ATTRIBUTE_LAST_CHECKED: str(data.remote_version_last_checked),
+        },
+    ),
+)
+
+
+class PaperlessBinarySensor(
+    PaperlessCoordinatorEntity[PaperlessCoordinator],
+    BinarySensorEntity,
+):
+    """Defines a Paperless-ngx binary sensor."""
+
+    entity_description: PaperlessBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PaperlessCoordinator,
+        data: PaperlessRuntimeData,
+        entry: PaperlessConfigEntry,
+        description: PaperlessBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize the binary sensor."""
+        super().__init__(data, entry, description, coordinator)
+        self.paperless_data = data
+        self._attr_unique_id = (
+            f"{DOMAIN}__{entry.entry_id}_binary_sensor_{description.key}"
```

Comment:
```suggestion
            f"{entry.entry_id}_{description.key}"
```
they are already unique per domain and platform